extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java

```diff
@@ -0,0 +1,284 @@
+    private static final int KAFKA_PORT = 9092;
+    private static final String KAFKA_BOOTSTRAP_SERVERS = "kafka.bootstrap.servers";
+
+    static volatile Closeable closeable;
+    static volatile KafkaDevServiceCfg cfg;
+    static volatile boolean first = true;
+
+    private final IsDockerWorking isDockerWorking = new IsDockerWorking(true);
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    public DevServicesKafkaBrokerBuildItem startKafkaDevService(
+            LaunchModeBuildItem launchMode,
+            KafkaBuildTimeConfig kafkaClientBuildTimeConfig,
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfiguration,
+            BuildProducer<ServiceStartBuildItem> serviceStartBuildItemBuildProducer) {
+
+        KafkaDevServiceCfg configuration = getConfiguration(kafkaClientBuildTimeConfig);
+
+        if (closeable != null) {
+            boolean shouldShutdownTheBroker = launchMode.getLaunchMode() == LaunchMode.TEST;
+            if (!shouldShutdownTheBroker) {
+                shouldShutdownTheBroker = !configuration.equals(cfg);
+            }
+            if (!shouldShutdownTheBroker) {
+                return null;
+            }
+            shutdownBroker();
+            cfg = null;
+        }
+
+        KafkaBroker kafkaBroker = startKafka(configuration);
+        DevServicesKafkaBrokerBuildItem bootstrapServers = null;
+        if (kafkaBroker != null) {
+            closeable = kafkaBroker.getCloseable();
+            runTimeConfiguration.produce(new RunTimeConfigurationDefaultBuildItem(
+                    KAFKA_BOOTSTRAP_SERVERS, kafkaBroker.getBootstrapServers()));
+            bootstrapServers = new DevServicesKafkaBrokerBuildItem(kafkaBroker.getBootstrapServers());
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = new Runnable() {
+                @Override
+                public void run() {
+                    if (closeable != null) {
+                        shutdownBroker();
+                    }
+                    first = true;
+                    closeable = null;
+                    cfg = null;
+                }
+            };
+            QuarkusClassLoader cl = (QuarkusClassLoader) Thread.currentThread().getContextClassLoader();
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(closeTask);
+            Thread closeHookThread = new Thread(closeTask, "Kafka container shutdown thread");
+            Runtime.getRuntime().addShutdownHook(closeHookThread);
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(new Runnable() {
+                @Override
+                public void run() {
+                    Runtime.getRuntime().removeShutdownHook(closeHookThread);
+                }
+            });
+        }
+        cfg = configuration;
+        return bootstrapServers;
+    }
+
+    private void shutdownBroker() {
+        if (closeable == null) {
```

Comment:
```suggestion
        if (closeable != null) {
```

???

Otherwise NPE is guaranteed.

Reply:
and now the question: why is this not called?